### PR TITLE
Use kernel dispatcher for symfony http cache

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Related issues/PRs | sulu/sulu#4124
| License | MIT

#### What's in this PR?

Use kernel dispatcher for symfony http cache.

#### Why?

It avoids errors that the server need to be able that it can handle real http request to itself. Also its a performance boost to use the kernel dispatcher.

Links:

 - https://github.com/FriendsOfSymfony/FOSHttpCache/blob/master/doc/proxy-clients.rst#kerneldispatcher-for-single-server-installations
 - https://foshttpcachebundle.readthedocs.io/en/latest/features/symfony-http-cache.html#optimization-for-single-server-installations
 - Example code I looked at: https://github.com/FriendsOfSymfony/FOSHttpCacheBundle/pull/491

### CODE REVIEW

File: public/website.php [hunk 1]
```diff
@@ -59,11 +59,7 @@
 // Comment this line if you want to use the "varnish" http
 // caching strategy. See http://sulu.readthedocs.org/en/latest/cookbook/caching-with-varnish.html
 if ($env !== 'dev') {
-    $kernel = new SuluHttpCache($kernel);
-
-    // When using the HttpCache, you need to call the method in your front controller
-    // instead of relying on the configuration parameter
-    Request::enableHttpMethodParameterOverride();
```
Comment (alexander-schranz) [inline public/website.php:66]: ~~could not find any reference to this in the foshttpcache docs so I removed  /cc @trickreich~~

Comment (alexander-schranz) [reply public/website.php:66]: found the reference in the symfony cache kernel docs so I readded it. I would always activate it to have no difference between request on dev or prod

**Comment by alexander-schranz** [comment]
ping @chirimoya @danrot

File: src/Kernel.php [hunk 2]
```diff
@@ -11,12 +11,23 @@
...
 
-class Kernel extends SuluKernel
+class Kernel extends SuluKernel implements HttpCacheProvider
 {
+    use HttpCacheAware;
+
+    public function __construct(string $environment, bool $debug, string $suluContext)
+    {
+        parent::__construct($environment, $debug, $suluContext);
+        $this->setHttpCache(new SuluHttpCache($this));
```
Comment (wachterjohannes) [inline src/Kernel.php:29]: comment for varnish

**Review by wachterjohannes — APPROVED** [review]

File: src/Kernel.php (hunk 2 above)
Comment (alexander-schranz) [reply src/Kernel.php:29]: done

**Pull request merged**